recipes/qt/6.x.x/conanfile.py

```diff
@@ -488,33 +489,74 @@ def build(self):
+                Documentation = res/datadir/doc
+                Examples = res/datadir/examples"""))
         self.copy("*LICENSE*", src="qt6/", dst="licenses")
         for module in self._submodules:
             if not self.options.get_safe(module):
                 tools.rmdir(os.path.join(self.package_folder, "licenses", module))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        for mask in ["Find*.cmake", "*Config.cmake", "*-config.cmake"]:
+            tools.remove_files_by_mask(self.package_folder, mask)
         tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la*")
         tools.remove_files_by_mask(self.package_folder, "*.pdb*")
         os.remove(os.path.join(self.package_folder, "bin", "qt-cmake-private-install.cmake"))
+        
+        for m in os.listdir(os.path.join(self.package_folder, "lib", "cmake")):
+            module = os.path.join(self.package_folder, "lib", "cmake", m, "%sMacros.cmake" % m)
+            if not os.path.isfile(module):
+                tools.rmdir(os.path.join(self.package_folder, "lib", "cmake", m))
+
+        extension = ""
+        if self.settings.os == "Windows":
+            extension = ".exe"
+        filecontents = "set(QT_CMAKE_EXPORT_NAMESPACE Qt6)\n"
+        ver = tools.Version(self.version)
+        filecontents += "set(QT_VERSION_MAJOR %s)\n" % ver.major
+        filecontents += "set(QT_VERSION_MINOR %s)\n" % ver.minor
+        filecontents += "set(QT_VERSION_PATCH %s)\n" % ver.patch
+        targets = ["moc", "rcc", "tracegen", "cmake_automoc_parser", "qlalr", "qmake"]
+        targets.extend(["qdbuscpp2xml", "qdbusxml2cpp"])
+        if self.options.gui:
+            targets.append("qvkgen")
+        if self.options.widgets:
+            targets.append("uic")
+        if self.options.qttools:
+            targets.extend(["qhelpgenerator", "qtattributionsscanner", "windeployqt"])
+            targets.extend(["lconvert", "lprodump", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
+        if self.options.qtshadertools:
+            targets.append("qsb")
+        if self.options.qtdeclarative:
+            targets.extend(["qmltyperegistrar", "qmlcachegen", "qmllint", "qmlimportscanner"])
+            targets.extend(["qmlformat", "qml", "qmlprofiler", "qmlpreview", "qmltestrunner"])
+        for target in targets:
+            filecontents += textwrap.dedent("""\
+                if(NOT TARGET ${{QT_CMAKE_EXPORT_NAMESPACE}}::{0})
+                    add_executable(${{QT_CMAKE_EXPORT_NAMESPACE}}::{0} IMPORTED)
+                    set_target_properties(${{QT_CMAKE_EXPORT_NAMESPACE}}::{0} PROPERTIES IMPORTED_LOCATION ${{CMAKE_CURRENT_LIST_DIR}}/../../../bin/{0}{1})
+                endif()
+                """.format(target, extension))
+        tools.save(os.path.join(self.package_folder, self._cmake_executables_file), filecontents)
```

Comment:
I'm not sure, but maybe the trick with `find_program` in protobuf recipe would also benefit to this recipe:
https://github.com/conan-io/conan-center-index/blob/493cd2694875f398494264bbceb90c1a196a8f3a/recipes/protobuf/all/conanfile.py#L120-L140

Reply:
Qt itself doesn't use find_program either. I think we should stick close to the Qt original approach unless we have a clear reason why something else is required. I think cross compilation might be a good one but I am not sure how that works with Qt, I think a lot more would need to happen. @ericLemanissier do you know? I think it might be a bit out of the scope of this PR (which has been open for 26 days already).

Module and bin more robust is a good one I think.